src/charm.py

```diff
@@ -110,6 +188,47 @@ def __init__(self, *args):
         self.framework.observe(
             self.on.validate_configuration_action, self._on_validate_configuration
         )
+        self.framework.observe(self.resources_patch.on.patch_failed, self._on_k8s_patch_failed)
+
+    @classmethod
```

Comment:
Does any client of this method have the need to use this _private method_ without an instance of the charm available, as in `PrometheusCharm._limits_to_requests()` ? If not, is there any reason for this to be a `classmethod` ? Could it just be a normal private instance method ?